src/DummyFileBasedCms.Web/Startup.cs

```diff
@@ -25,24 +19,37 @@ public Startup( IConfiguration configuration )
 
         public IConfiguration Configuration { get; }
 
-        // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices( IServiceCollection services )
         {
-            services.Configure<FlatFileCmsGitOptions>( Configuration.GetSection( "FlatFileCmsGit" ) );
+            // Contruct an options instance so we can use it to configure things in startup.
+            var cmsOptions = new FlatFileCmsGitOptions();
+            Configuration.Bind("FlatFileCmsGit", cmsOptions);
+
+            // Bind the configuration in DI so it's accessible to services.
+            services.Configure<FlatFileCmsGitOptions>("FlatFileCmsGit", Configuration);
 
-            var flatFileCmsOptions = services.BuildServiceProvider().GetService<IOptions<FlatFileCmsGitOptions>>()?.Value;
-            var fileProvider = new PhysicalFileProvider(flatFileCmsOptions.FilePath);
+            // Adding a changing file providers here changes were MVC looks for views/pages. If you need
+            // to get the file provider(s) from DI you can resolve IRazorViewEngineFileProviderAccessor from
+            // DI
+            services.Configure<RazorViewEngineOptions>(options =>
+            {
+                // You'll need to decide if you want to include files in the application's directory
+                // or not.
+                // options.FileProviders.Clear();
 
-            services.AddSingleton(fileProvider);
-            services.AddTransient<FlatFileCmsProviderRazorProjectFileSystem>();
-            services.AddSingleton<IPageRouteModelProvider, FlatFileCmsRazorProjectPageRouteModelProvider>();
+                // File Providers throw if the directory doesn't exist. 
+                if (Directory.Exists(cmsOptions.FilePath))
+                {
+                    options.FileProviders.Add(new PhysicalFileProvider(cmsOptions.FilePath));
+                }
+            });
```

Comment:
This is all that's required to get MVC to look in another directory - if you need to change directories at runtime, thats a case where you might need to implement your own file provider.